### PR TITLE
Add Host Authorization checkbox for Property Managers

### PR DESCRIPTION
**Pull request by dimak1**
*Issue:*

- bcgov/entity/issues/23745

*Description of changes:*
- On Review and Confirm page - add a checkbox when Property Manager exists
- Add validation, test update

Screenshot 2024-10-29 at 08 49 14

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

**Comment by shaangill025** [comment]
@dimak1 Should the checkbox still be visible if it is property host and selected yes for adding property manager info?

**Comment by dimak1** [comment]
> @dimak1 Should the checkbox still be visible if it is property host and selected yes for adding property manager info?

@shaangill025 hmm I check only for `hasPropertyManager` flag. @andyyanggov could you please confirm when is the checkbox is shown? Thanks

**Comment by andyyanggov** [comment]
> @andyyanggov could you please confirm when is the checkbox is shown? Thanks

It's intended for the PM to declare that they're authorized to do this on the host's behalf, so it should be hidden if the host themself is filling out the form (regardless if they hire a PM or not).

**Comment by dimak1** [comment]
@shaangill025 based on Andy's answer, the checkbox is hidden correctly. What you think?

**Comment by shaangill025** [comment]
> based on Andy's answer, the checkbox is hidden correctly. What you think?

With the preview url, I see the checkbox:
image
image

**Comment by dimak1** [comment]
@shaangill025 ok i see the issue now, thanks for catching it.

**Comment by dimak1** [comment]
@shaangill025 updated, thanks!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

File: strr-web/pages/create-account.vue [hunk 1]
```diff
@@ -206,7 +206,7 @@ const validateStep = (schema: any, state: any, index: number) => {
 }
 
 const validatePropertyManagerStep = () => {
-  if (!formState.isPropertyManagerRole && !formState.hasPropertyManager) {
+  if (!formState.isPropertyManagerRole) {
```
Comment (shaangill025) [inline strr-web/pages/create-account.vue:209]: Stepper validation is breaking with this formData
image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

File: strr-web/components/bcros/form-section/property/Form.vue [hunk 2]
```diff
@@ -96,7 +96,7 @@ watch(canadaPostAddress, (newAddress) => {
 
 watch(() => formState.propertyDetails.isUnitOnPrincipalResidenceProperty, (newValue) => {
   if (!newValue) {
-    formState.propertyDetails.hostResidence = null // Reset if not required
+    formState.propertyDetails.hostResidence = undefined // Reset if not required
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/property/Form.vue:99]: TS error fix

File: strr-web/interfaces/account-i.ts [hunk 3]
```diff
@@ -175,7 +175,8 @@ export interface CreateAccountFormStateI {
   }
   selectedAccount: OrgI
   principal: PrincipalResidenceI
-  supportingDocuments: File[]
+  supportingDocuments: File[],
+  hasHostAuthorization: boolean
```
Comment (dimak1) [inline strr-web/interfaces/account-i.ts:179]: @shaangill025 if backend needs it as some point, we can easily pass it.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

File: strr-web/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "strr-web",
-  "version": "0.1.42",
+  "version": "0.1.44",
```
Comment (shaangill025) [inline strr-web/package.json:3]: Shouldn't it be 43.

File: strr-web/pages/create-account.vue (hunk 1 above)
Comment (dimak1) [reply strr-web/pages/create-account.vue]: Fixed

File: strr-web/package.json (hunk 4 above)
Comment (dimak1) [reply strr-web/package.json]: Fixed. Not sure, maybe typo. Thanks!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

**Review by shaangill025 — APPROVED** [review]
LGTM

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-239-uxhsc5yl.web.app

**Pull request merged**